tests/Doctrine/Tests/ORM/Hydration/AbstractHydratorTest.php

```diff
@@ -106,4 +107,31 @@ public function testHydrateAllRegistersAndClearsAllAttachedListeners(): void
 
         $this->hydrator->hydrateAll($this->mockStatement, $this->mockResultMapping);
     }
+
+    /**
+     * @group #DDC-8482
+     */
+    public function testHydrateAllClearsAllAttachedListenersEvenOnError(): void
+    {
+        $this
+            ->mockEventManager
+            ->expects(self::at(0))
```

Comment:
`self::at` is deprecated in recent versions of PHPUnit, please avoid it

Reply:
I've changed it and rebased the MR although I did not find a clean replacement for the self::at method.

I want to keep the check of the method call order so please look into it and say if it seems ok to you.

I can update the other methods in the test as well although it's out of scope of this MR so I'm not sure if I should :)

Reply:
Fixed and rebased, although I do not like the replacement (I did not find any other recommended way). Say if it seems ok to you please. I might replace the the rest of self::at occurrences in this Test class if requested (although it's out of scope of this MR)

Reply:
> I might replace the the rest of self::at occurrences in this Test class if requested (although it's out of scope of this MR)

It would certainly be appreciated, but it's not at all required.